### PR TITLE
ricochet-refresh: init at 3.0.18

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

Secure chat without DNS or WebPKI.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
- [x] Tested basic functionality of all binary files
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by amjoseph-nixpkgs** [comment]
> Needs adding to all-packages?

Nope!  Now that we have `pkgs/by-name` it's all automatic now.

- https://github.com/NixOS/rfcs/pull/140

**Review by hlandau — APPROVED** [review]
Looks good to me.

**Comment by amjoseph-nixpkgs** [comment]
Bad ofborg!

```
these 2 derivations will be built:
  /nix/store/dj2bfw0hicwzp9r715z3ivlzmp6g7kdn-all-attrs-eval-under-tryEval.drv
  /nix/store/80d4mn25kq78avf1fxk7239cgrl6zgjw-nixpkgs-lib-tests.drv
building '/nix/store/dj2bfw0hicwzp9r715z3ivlzmp6g7kdn-all-attrs-eval-under-tryEval.drv'...
cp: cannot create regular file 'pkgs/development/python-modules/face-recognition/models.nix': No space left on device
cp: cannot create directory 'pkgs/development/python-modules/facebook-sdk': No space left on device
cp: cannot create directory 'pkgs/development/python-modules/facedancer': No space left on device
cp: cannot create directory 'pkgs/development/python-modules/facenet-pytorch': No space left on device
cp: cannot create directory 'pkgs/development/python-modules/factory-boy': No space left on device
```

**Comment by amjoseph-nixpkgs** [comment]
@ofborg eval

**Comment by amjoseph-nixpkgs** [comment]
Well apparently no CI today.  I'll try again tomorrow.

File: pkgs/by-name/ri/ricochet-refresh/package.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      The original Ricochet uses onion "v2" hashed-RSA addresses,
+      which are no longer supported by the Tor network.  Ricochet
+      Refresh upgrades the original Ricochet protocol to use the
+      current onion "v3" ed25519 addresses.
+    '';
+    homepage = "https://www.ricochetrefresh.net/";
+    downloadPage = "https://github.com/blueprint-freespeech/ricochet-refresh/releases";
+    license = lib.licenses.bsd3;
+  };
+
```
Comment (AndersonTorres) [inline pkgs/by-name/ri/ricochet-refresh/package.nix:80]:
```suggestion
```

Comment (AndersonTorres) [inline pkgs/by-name/ri/ricochet-refresh/package.nix:80]:
```suggestion
```

File: pkgs/by-name/ri/ricochet-refresh/package.nix [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    qt5.wrapQtAppsHook
+  ];
+
+  enableParallelBuilding = true;
+
+  # https://github.com/blueprint-freespeech/ricochet-refresh/blob/main/BUILDING.md
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=MinSizeRel"
+    "-DRICOCHET_REFRESH_INSTALL_DESKTOP=ON"
+    "-DUSE_SUBMODULE_FMT=ON"
```
Comment (AndersonTorres) [inline pkgs/by-name/ri/ricochet-refresh/package.nix:55]:
```suggestion
    (lib.cmakeFeature "CMAKE_BUILD_TYPE" "MinSizeRel")
    (lib.cmakeBool "RICOCHET_REFRESH_INSTALL_DESKTOP" true)
    (lib.cmakeBool "USE_SUBMODULE_FMT" true)
```

File: pkgs/by-name/ri/ricochet-refresh/package.nix [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  src = fetchFromGitHub {
+    owner = "blueprint-freespeech";
+    repo = pname;
+    rev = "v${version}-release";
+    sha256 = "sha256-QN2cxcYWGoszPdrWv+4FoTGNjQViK/OwxbBC6uoDhfA=";
+    fetchSubmodules = true;
+  };
+
+  sourceRoot = "source/src";
```
Comment (AndersonTorres) [inline pkgs/by-name/ri/ricochet-refresh/package.nix:28]:
```suggestion
  sourceRoot = "${finalAttrs.src.name}/src";
```

File: pkgs/by-name/ri/ricochet-refresh/package.nix [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+, qt5
+, openssl
+, protobuf3_20  # https://github.com/blueprint-freespeech/ricochet-refresh/issues/178
+, pkg-config
+, cmake
+}:
+
+let
+  pname = "ricochet-refresh";
+  version = "3.0.18";
```
Comment (AndersonTorres) [inline pkgs/by-name/ri/ricochet-refresh/package.nix:13]:
```suggestion
```

File: pkgs/by-name/ri/ricochet-refresh/package.nix [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  protobuf = protobuf3_20;
+in
+stdenv.mkDerivation (finalAttrs: {
+
+  inherit pname version;
```
Comment (AndersonTorres) [inline pkgs/by-name/ri/ricochet-refresh/package.nix:18]:
```suggestion
  pname = "ricochet-refresh";
  version = "3.0.18";
```

File: pkgs/by-name/ri/ricochet-refresh/package.nix [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+  src = fetchFromGitHub {
+    owner = "blueprint-freespeech";
+    repo = pname;
+    rev = "v${version}-release";
+    sha256 = "sha256-QN2cxcYWGoszPdrWv+4FoTGNjQViK/OwxbBC6uoDhfA=";
```
Comment (AndersonTorres) [inline pkgs/by-name/ri/ricochet-refresh/package.nix:24]:
```suggestion
    repo = "ricochet-refresh";
    rev = "v${finalAttrs.version}-release";
    hash = "sha256-QN2cxcYWGoszPdrWv+4FoTGNjQViK/OwxbBC6uoDhfA=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by hlandau — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Comment by amjoseph-nixpkgs** [comment]
https://github.com/NixOS/nixpkgs/issues/50105

**Pull request merged**